packages/lesswrong/components/ultraFeed/UltraFeedPostDialog.tsx

```diff
@@ -450,123 +534,216 @@ const UltraFeedPostDialog = ({
   useModalHashLinkScroll(scrollableContentRef, true, true, (footnoteHTML: string) => {
     setFootnoteDialogHTML(footnoteHTML);
   });
 
-  // Bridge scroll events from internal container to window so hooks relying on window scroll keep working
-  useEffect(() => {
-    const el = scrollableContentRef.current;
-    if (!el) return;
-    const handler = () => {
-      window.dispatchEvent(new Event('scroll'));
-    };
-    el.addEventListener('scroll', handler);
-    return () => {
-      el.removeEventListener('scroll', handler);
-    };
-  }, [hasTocData]);
-
-  // Handle clicking on comment count to scroll to comments
+  const toggleEmbeddedPlayer = displayPost && postHasAudioPlayer(displayPost) ? (e: React.MouseEvent) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const action = showEmbeddedPlayer ? "close" : "open";
+    const newCookieValue = showEmbeddedPlayer ? "false" : "true";
+    captureEvent("toggleAudioPlayer", { action, pageModalContext: "ultraFeedPostModal" });
+    setCookie(SHOW_PODCAST_PLAYER_COOKIE, newCookieValue, { path: "/" });
+    setShowEmbeddedPlayer(!showEmbeddedPlayer);
+  } : undefined;
+  
   const scrollToComments = (e: React.MouseEvent) => {
     e.preventDefault();
     e.stopPropagation();
-    
     captureEvent("ultraFeedPostDialogScrollToComments");
     
     const container = scrollableContentRef.current;
-    // Look for the comments section wrapper which always exists
+    if (displayPost.question) {
+      const answersElement = document.getElementById('answers');
+      if (container && answersElement) {
+        scrollToElementInContainer(container, answersElement);
+        return;
+      }
+    }
+    
     const commentsElement = document.getElementById('commentsSection');
     if (container && commentsElement) {
       scrollToElementInContainer(container, commentsElement);
     }
   };
+  
+  // Reset scroll tracking when target changes
+  useEffect(() => {
+    hasScrolledRef.current = false;
+  }, [targetCommentId]);
+  
+  // Scroll to target comment when loaded
+  useEffect(() => {
+    let fadeTimer: NodeJS.Timeout | null = null;
+
+    if (!isCommentsLoading && targetCommentId && comments && comments.length > 0 && !hasScrolledRef.current) {
+      const targetFound = comments.some(c => c._id === targetCommentId);
+      
+      if (targetFound) {
+        const cleanup = scrollToElement(targetCommentId, () => {
+          const element = document.getElementById(targetCommentId);
+          if (element) {
+            // Add highlight animation
+            element.classList.add(classes.scrolledHighlight);
+
+            fadeTimer = setTimeout(() => {
+              element.classList.add(classes.scrolledHighlightFading);
+            }, 100);
+
+            // Mark that we've scrolled
+            hasScrolledRef.current = true;
+          }
+        });
+
+        return () => {
+          cleanup();
+          if (fadeTimer) clearTimeout(fadeTimer);
+        };
+      }
+    }
```

Comment:
idk what any of this is doing and am worried that it's not particularly necessary or justifying the complexity

Reply:
this is pre-existing and necessary (I just moved it around)

- there's highlighting that fades when you get scrolled to a comment
- we want you to get scrolled when you open the modal but only once, we track that